reporter/src/web-app/js/src/components/TreeView.js

```diff
@@ -32,126 +36,229 @@ class TreeView extends React.Component {
+      super(props);
+      this.state = { search: '' };
+      if (props.reportData && props.reportData.tree) {
+          this.state = {
+              ...this.state,
+              tree: props.reportData.tree,
+              expandedKeys: [],
+              selectedIndex: 0,
+              sorted: [],
+          };
+      }
+  }
 
-    onSearchChange = (e) => {
-        const { tree } = this.state;
-        const searchVal = e.target.value;
-        const reccurSearch = (arr, parent, search) => arr.reduce((acc, item) => {
-            const matches = item.name.indexOf(search) >= 0;
-            let keys = [];
-            if (item.children) {
-                keys = reccurSearch(item.children, item, search);
-            }
-            if (matches) {
-                keys.push(parent.id || parent.name);
-            }
-            return [...acc, ...keys];
-        }, []);
-        if (tree) {
-            const keysToExpand = searchVal ? reccurSearch(tree, { key: 'root' }, searchVal) : [];
-            this.setState({
-                expandedKeys: keysToExpand,
-                autoExpandParent: true,
-                search: searchVal,
-            });
-        }
-    }
+  onExpand = (expandedKeys) => {
+      this.setState({
+          expandedKeys,
+          autoExpandParent: false,
+      });
+  }
 
-    renderTreeNode = (treeLeaf) => {
-        const { name } = treeLeaf;
-        const { search } = this.state;
-
-        let nameMapped = (
-            <span>
-                {name}
-            </span>
-        );
-
-        if (search && name.indexOf(search) >= 0) {
-            const index = name.indexOf(search);
-            const pre = name.substring(0, index);
-            const post = name.substring(index + search.length);
-            nameMapped = (
-                <span>
-                    {pre}
-                    <span
-                        style={{
-                            color: 'red',
-                        }}
-                    >
-                        {search}
-                    </span>
-                    {post}
-                </span>
-            );
-        }
-
-        const tooltip = (
-            treeLeaf.path.length >= 1 && (
-                <div className="tooltip-list">
-                    <div class-name="tooltip-item">
-                        {treeLeaf.path.slice(0).join(' / ')}
-                    </div>
-                </div>
-            )
-        );
-
-        const nameWithTooltip = (
-            <Tooltip overlayClassName="oss-tree-tooltip" placement="right" title={tooltip}>
-                {nameMapped}
-            </Tooltip>
-        );
-
-        return (
-            <TreeNode
-                key={treeLeaf.id || treeLeaf.name}
-                title={nameWithTooltip}
-            >
-                {treeLeaf.children.map(childLeaf => this.renderTreeNode(childLeaf))}
-            </TreeNode>
-        );
-    }
+  componentDidUpdate(_, prevState) {
```

Comment:
Fix indentation through out TreeView.js to be 4 spaces not 2

Reply:
+